lib/index.d.ts

```diff
@@ -817,6 +828,14 @@ interface NamedSubscription {
     timeToLive: number;
 }
 
+type ExtractPropertyNamesOfType<T, S> = {
+    [K in keyof T]: T[K] extends S ? K : never
+}[keyof T];
+
+type RealmPartialModel<T> =
```

Comment:
I don't know if `Model` is the right word here - we're already using `ObjectSchema` and `Class` elsewhere.

Reply:
This has now changed name to `RealmInsertionModel` I personally think it's quite descriptive and a commonly used name. I don't the either Schema or Class fits here, but could use `RealmInsertionObject`?
This naming is already in v10, so I suggest we revisit the naming if it's confusing to users?